src/test/java/picard/util/help/DocumentationGenerationIntegrationTest.java

```diff
@@ -0,0 +1,79 @@
+package picard.util.help;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.spi.ToolProvider;
+
+/**
+ * Smoke test to run doc gen on a subset of classes to make sure it doesn't regress.
+ */
+public class DocumentationGenerationIntegrationTest {
+    /**
+     * Entry point for manually running the picardDoc process on a subset of packages from within picard.
+     */
+    private static String[] docTestPackages = {
+            // note that the fingerprint package can't be in this list, since it requires the google
+            // cloud nio library to be present to compile
+            "picard.analysis",
+            "picard.metrics"
+    };
+
+    // suppress deprecation warning on Java 11 since we're using deprecated javadoc APIs
+    @Test
+    public static void documentationSmokeTest() throws IOException {
+        final Path docTestTarget =  Files.createTempDirectory("docgentest");
+        final String[] argArray = new String[]{
+                //"javadoc",
+                "-doclet", PicardHelpDoclet.class.getName(),
+                "-docletpath", System.getProperty("java.class.path"),
+                "-sourcepath", "src/main/java",
+                "-settings-dir", "src/main/resources/picard/helpTemplates",
+                "-d", docTestTarget.toFile().getAbsolutePath(), // directory must exist
+                "-output-file-extension", "html",
+                "-build-timestamp", "2016/11/11 11:11:11",
+                "-absolute-version", "1.1-111",
+                "-verbose"
+        };
+
+        final List<String> docArgList = new ArrayList<>();
+        docArgList.addAll(Arrays.asList(argArray));
+        docArgList.add("-cp");
+        docArgList.add(System.getProperty("java.class.path"));
+        docArgList.addAll(Arrays.asList(docTestPackages));
+
+        // Run javadoc in the current JVM with the custom doclet, and make sure it succeeds (this is a smoke test;
+        // we just want to make sure it doesn't blow up).
+        ToolProvider jdProvider = null;
```

Comment:
Would it be cleaner to use ToolProvider.findFirst("javadoc")?
```
ToolProvider jdProvider = ToolProvider.findFirst("javadoc")
                .orElseThrow(() ->  new IllegalStateException("Can't find javadoc tool"));
``` ?

Reply:
Yeah, better. I need to remember to do this in GATK as well.